### DISCUSSION TITLE
keycloak 25 unable to connect with another pod using service name

### DISCUSSION THREAD

**Discussion by Ajmalkhan7** [keycloak/keycloak#32774] [Archive]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

I have keycloak codecentric helm chart deployed and keycloak version is 25.0.0 and its up and running. I want to connect springboot application with keycloak.

But i am getting the following error

Couldn't retrieve remote JWK set: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "https://example.com/auth/realms/RELAM_NAME/protocol/openid-connect/certs": PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

My application.yaml configurations for keycloak are as follows:

keycloak:
jwk-set-uri: https://example.com/auth/realms/REALM_NAME/protocol/openid-connect/certs
auth-server-url: https://example.com/

keycloak-tenants:
auth-server-url: https://example.com/
realms:
- CLI_ADMIN:
realm: REL_NAME
resource: CLI_ADMIN
principal-attribute:
admin-cli: false
client-secret: M
order: 1
- CLI_MASTER:
realm:
resource:
principal-attribute:
admin-cli: false
client-secret:
order: 2

security:
jwt:
issuers:
- https://example.com/auth/realms/REALM_NAME
- https://example.com/auth/realms/REALM_NAME

and values.yaml for code centric helm charts is as follows

command:

"/opt/keycloak/bin/kc.sh"
"start"
"--spi-events-listener-jboss-logging-success-level=info"
"--spi-events-listener-jboss-logging-error-level=warn"
extraEnv: |

name: KEYCLOAK_ADMIN
value: admin
name: KEYCLOAK_ADMIN_PASSWORD
value: admin
name: JAVA_OPTS_APPEND
value: >-
-Djgroups.dns.query={{ include "keycloak.fullname" . }}-headless
-Dkeycloak.hostname=example.com
-Dkeycloak.proxy=forward
name: KC_HOSTNAME_STRICT
value: "false"
name: KC_HTTP_RELATIVE_PATH
value: "/auth"
name: KC_CACHE_STACK
value: kubernetes
name: KC_PROXY
value: edge
name: HTTP_ADDRESS_FORWARDING
value: "true"
name: KC_HOSTNAME
value: example.com
name: KC_HEALTH_ENABLED
value: "true"
name: KC_HTTP_ENABLED # SSL termnites at reverser proxy, need this enabled.
value: "true"
name: KC_HOSTNAME_STRICT_HTTPS # SSL termnites at reverser proxy, need this disabled.
value: "true"
name: KC_SPI_HOSTNAME_DEFAULT_ADMIN
value: "example.com"
name: PROXY_ADDRESS_FORWARDING
value: "true"
name: KEYCLOAK_PRODUCTION
value: "true"
Any help on this would be highly appreciated.

### Version

codecentric helm charts 25.0.0

### Regression

- [ ] The issue is a regression

### Expected behavior

spring boot should retrive the token from keycloak

### Actual behavior

Its throwing error as above

### How to Reproduce?

Install helm chart 25.0.0 of codecentric

**Comment by shawkins** [comment]
Converted to a discussion instead of an issue. 

> Couldn't retrieve remote JWK set: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "https://example.com/auth/realms/RELAM_NAME/protocol/openid-connect/certs": PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

Givne that you are running in edge mode such that ssl requests are not reaching the keycloak server, your problem lies either with your proxy or your spring boot client. The error message typically means that the client does not trust the server. Make sure that the client truststore includes the cert for example.com.

**Reply by Ajmalkhan7** [reply]
I have already added .crt in my sprinboot app and removed proxy edge but still getting same error. 
Any other thing do i need to check?

**Reply by darius-m** [reply]
Do connections to Keycloak using other tools (e.g. `curl`) from the application container work? This should help identifying if the issue is application-specific or has something to do with the container / Keycloak configuration.
